### DISCUSSION TITLE
Do packages on DefinitelyTyped count as derivative works?

### DISCUSSION THREAD

**Discussion by MysteryBlokHed** [DefinitelyTyped/DefinitelyTyped#57232] [_Misc]
I'm talking to a maintainer of an open-source JavaScript project licensed under LGPL-3.0, who says that the types I've made should be licensed under the original project's license. Is this true?
I'm not familiar with copyright law and couldn't find other instances of this happening when I looked through past issues/PR's for this repo.

**Comment by MysteryBlokHed** [comment]
Relevant PR's: #57016 (added types) #57221 (owner requesting license change)

A DefinitelyTyped maintainer said they're consulting lawyers about this, so I should have an answer then.

**Comment by RyanCavanaugh** [comment]
This is not a legal opinion from me nor our legal team (they have not gotten back to me), but I will provide some information.

https://copyleft.org/guide/comprehensive-gpl-guidech5.html

> In the case of Oracle America v. Google, 872 F. Supp.2d 974 (N.D. Cal. 2012), the Northern District of California District Court examined the question of whether the application program interfaces (APIs) associated with the Java programming language are entitled to copyright protection. While the court expressly declined to rule whether all APIs are free to use without license (872 F. Supp.2d 974 at 1002), **the court held that the command structure and taxonomy of the APIs were not protectable under copyright law**.

(emphasis mine)
